freshmaker/kojiservice.py

```diff
@@ -407,6 +407,41 @@ def get_bundle_csv(self, build_nvr):
         with zipfile.open(csv_files[0]) as f:
             return yaml.safe_load(f)
 
+    @region.cache_on_arguments()
+    def get_module_name_streams(self, build_nvr):
+        """
+        Get module's name and its streams through build NVR
+
+        :param str build_nvr: NVR of module build.
+        :return: a dict for module name and its streams, e.g. {"key": [values]}
+        :rtype: dict
+        """
+        module_name_streams_mapping = dict()
+        build = self.get_build(build_nvr)
+        modulemd_str = build.get("extra", {}).get("typeinfo", {}).get("module", {}).get("modulemd_str")
+        if modulemd_str:
+            try:
+                import gi
+                gi.require_version("Modulemd", "2.0")
+                from gi.repository import Modulemd
+                fedora_index = Modulemd.ModuleIndex.new()
```

Comment:
By checking the API doc, `Modulemd.ModuleIndex` [1] is mainly designed for dealing with repodata, we only need to deal with a modulemd string, so I think using `Modulemd.ModuleStream` [2] could be better.

[1] https://fedora-modularity.github.io/libmodulemd/latest/modulemd-2.0-Modulemd.ModuleIndexMerger.html
[2] https://fedora-modularity.github.io/libmodulemd/latest/ModulemdModuleStream.html

Reply:
Ack. Fixing